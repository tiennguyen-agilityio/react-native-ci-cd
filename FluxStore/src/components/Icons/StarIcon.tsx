import {memo} from 'react';
import Svg, {Path} from 'react-native-svg';
import {TouchableOpacity} from 'react-native';

import {IconProps} from '@/interfaces';
import {colors} from '@/themes';

export const StarIcon = memo(
  ({
    width = 9,
    height = 8,
    disabled = false,
    isActive = false,
    color = colors.green[200],
    style,
    onPress,
  }: IconProps) => (
    <TouchableOpacity
      testID="star-icon"
      activeOpacity={0.8}
      onPress={onPress}
      disabled={disabled}
      style={style}>
      <Svg width={width} height={height} viewBox="0 0 9 8" fill="none">
        {isActive ? (
          <Path
            fillRule="evenodd"
            clipRule="evenodd"
            d="M6.097 2.624a.328.328 0 01-.313-.22L5.058.22a.332.332 0 00-.625 0l-.73 2.18a.328.328 0 01-.313.22H1.025a.32.32 0 00-.311.216.313.313 0 00.117.357l1.92 1.359a.31.31 0 01.118.352L2.137 7.1a.315.315 0 00.166.383.331.331 0 00.34-.03l1.907-1.35a.339.339 0 01.389 0l1.906 1.349a.331.331 0 00.34.03.326.326 0 00.178-.232.315.315 0 00-.011-.15L6.62 4.906a.311.311 0 01.118-.352L8.658 3.2a.313.313 0 00-.002-.516.321.321 0 00-.191-.057l-2.368-.003z"
            fill={color}
          />
        ) : (
          <Path
            d="M4.31.25c.017 0 .035.005.048.015a.083.083 0 01.018.017l.012.023.724 2.178.001.003a.578.578 0 00.55.387v.001l2.367.003h.006a.07.07 0 01.042.013c.012.008.021.02.026.033a.064.064 0 010 .04l-.023.032h0L6.16 4.35a.561.561 0 00-.234.548l.02.087.001.001.732 2.194c.002.009.003.019.002.028l-.013.028a.076.076 0 01-.026.023l-.036.007h-.003a.082.082 0 01-.025-.004l-.022-.012h-.002l-1.907-1.35h0a.589.589 0 00-.598-.046l-.078.046H3.97l-1.907 1.35h-.002a.082.082 0 01-.022.012l-.024.004h-.004a.076.076 0 01-.034-.007l-.028-.023a.065.065 0 01-.012-.028.064.064 0 01.002-.03v-.001l.732-2.192v-.002a.561.561 0 00-.212-.634h0L.542 2.989H.54l-.024-.033a.063.063 0 010-.038.072.072 0 01.026-.035l.042-.013h2.371v0a.578.578 0 00.513-.305l.036-.082V2.48l.73-2.175a.084.084 0 01.03-.04.082.082 0 01.022-.011L4.311.25z"
            stroke={color}
            strokeWidth={0.5}
          />
        )}
      </Svg>
    </TouchableOpacity>
  ),
);
