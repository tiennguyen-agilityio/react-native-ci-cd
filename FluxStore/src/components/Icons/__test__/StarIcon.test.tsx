import {render, fireEvent, screen} from '@testing-library/react-native';

import {StarIcon} from '../StarIcon';

describe('StarIcon', () => {
  const onPress = jest.fn();

  it('renders correctly (matches snapshot)', () => {
    const {toJSON} = render(<StarIcon />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly (matches snapshot) with isActive', () => {
    const {toJSON} = render(<StarIcon isActive />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls onPress when touched and not disabled', () => {
    render(<StarIcon onPress={onPress} />);

    fireEvent.press(screen.getByTestId('star-icon'));

    expect(onPress).toHaveBeenCalledTimes(1);
  });

  it('does NOT call onPress when disabled', () => {
    render(<StarIcon onPress={onPress} disabled />);

    fireEvent.press(screen.getByTestId('star-icon'));

    expect(onPress).not.toHaveBeenCalledTimes(0);
  });
});
