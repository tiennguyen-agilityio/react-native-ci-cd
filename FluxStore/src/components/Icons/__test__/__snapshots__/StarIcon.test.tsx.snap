// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarIcon renders correctly (matches snapshot) 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="star-icon"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={8}
    bbWidth={9}
    fill="none"
    focusable={false}
    height={8}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 8,
          "width": 9,
        },
      ]
    }
    vbHeight={8}
    vbWidth={9}
    width={9}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M4.31.25c.017 0 .035.005.048.015a.083.083 0 01.018.017l.012.023.724 2.178.001.003a.578.578 0 00.55.387v.001l2.367.003h.006a.07.07 0 01.042.013c.012.008.021.02.026.033a.064.064 0 010 .04l-.023.032h0L6.16 4.35a.561.561 0 00-.234.548l.02.087.001.001.732 2.194c.002.009.003.019.002.028l-.013.028a.076.076 0 01-.026.023l-.036.007h-.003a.082.082 0 01-.025-.004l-.022-.012h-.002l-1.907-1.35h0a.589.589 0 00-.598-.046l-.078.046H3.97l-1.907 1.35h-.002a.082.082 0 01-.022.012l-.024.004h-.004a.076.076 0 01-.034-.007l-.028-.023a.065.065 0 01-.012-.028.064.064 0 01.002-.03v-.001l.732-2.192v-.002a.561.561 0 00-.212-.634h0L.542 2.989H.54l-.024-.033a.063.063 0 010-.038.072.072 0 01.026-.035l.042-.013h2.371v0a.578.578 0 00.513-.305l.036-.082V2.48l.73-2.175a.084.084 0 01.03-.04.082.082 0 01.022-.011L4.311.25z"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        propList={
          [
            "stroke",
            "strokeWidth",
          ]
        }
        stroke={
          {
            "payload": 4283468411,
            "type": 0,
          }
        }
        strokeWidth={0.5}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`StarIcon renders correctly (matches snapshot) with isActive 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="star-icon"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={8}
    bbWidth={9}
    fill="none"
    focusable={false}
    height={8}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 8,
          "width": 9,
        },
      ]
    }
    vbHeight={8}
    vbWidth={9}
    width={9}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        clipRule={0}
        d="M6.097 2.624a.328.328 0 01-.313-.22L5.058.22a.332.332 0 00-.625 0l-.73 2.18a.328.328 0 01-.313.22H1.025a.32.32 0 00-.311.216.313.313 0 00.117.357l1.92 1.359a.31.31 0 01.118.352L2.137 7.1a.315.315 0 00.166.383.331.331 0 00.34-.03l1.907-1.35a.339.339 0 01.389 0l1.906 1.349a.331.331 0 00.34.03.326.326 0 00.178-.232.315.315 0 00-.011-.15L6.62 4.906a.311.311 0 01.118-.352L8.658 3.2a.313.313 0 00-.002-.516.321.321 0 00-.191-.057l-2.368-.003z"
        fill={
          {
            "payload": 4283468411,
            "type": 0,
          }
        }
        fillRule={0}
        propList={
          [
            "fill",
            "fillRule",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
